src/pudl/io_managers.py

```diff
@@ -536,9 +638,19 @@ def load_input(self, context: InputContext) -> pd.DataFrame:
 
 
 @io_manager
-def pudl_sqlite_io_manager(init_context) -> PudlSQLiteIOManager:
+def pudl_sqlite_io_manager(init_context) -> IOManager:
```

Comment:
@bendnorman and @zaneselvans - given that this now returns mixed format IO manager, could we rename this resource/io-manager from `pudl_sqlite_io_manager` to plain `pudl_io_manager`? I think that would make it more appropriate. There's a bit of a risk that very widely used word `pudl` refers both to *the primary output database* (`pudl.sqlite` up to this point) as well as the overall project/ETL/codebase. This ambiguity may be okay, because I'm not quite sure what would be the other word we could substitute here. Maybe `main_output` or something along those lines?

Reply:
I think `pudl_io_manager` is appropriate for this IO Manager. We're currently ok with the naming ambiguity of the code and data outputs. We explain the different parts of pudl in the [readme](https://github.com/catalyst-cooperative/pudl?tab=readme-ov-file#what-is-pudl).

Reply:
I think `pudl_io_manager` makes sense.

This is probably beyond the scope of this PR and maybe more of a @bendnorman thing but I think we should also change the way the `default_resources` are defined in `pudl.etl.__init__.py`:

```py
default_resources = {
    "datastore": datastore,
    "pudl_sqlite_io_manager": pudl_sqlite_io_manager,
    "ferc1_dbf_sqlite_io_manager": ferc1_dbf_sqlite_io_manager,
    "ferc1_xbrl_sqlite_io_manager": ferc1_xbrl_sqlite_io_manager,
    "dataset_settings": dataset_settings,
    "ferc_to_sqlite_settings": ferc_to_sqlite_settings,
    "epacems_io_manager": epacems_io_manager,
}
```

It seems like the keys in this dictionary should not specify anything about the format / destination of the data, while the values (the actual IO managers) should have that specificity.  So `pudl_io_manager` could be any kind of IO manager... but regardless it's the one that's going to be used to output PUDL data, whether it's going to SQLite, BigQuery, both SQLite + Parquet, etc.  And the value associated with it determines the details.

I found this very confusing in thinking about how we were going to switch over to writing the data to not just SQLite, since we've hard-coded `pudl_sqlite_io_manager` all throughout the system.  But it's not really an SQLite IO Manager. It could be any kind of IO Manager.  But I didn't realize they were separable until reviewing this PR.

The FERC IO managers seem like they have a similar issue, and also, what IO Managers are used to output all of the FERC Forms that aren't Form 1?

Reply:
I agree Zane the keys shouldn't specify anything about format / destination. 

The FERC IO managers are just for reading data out of the dbf and xbrl sqlite databases. We don't process any of the other ferc forms in the main PUDL ETL so we don't have IO managers for them.

Reply:
Also don't we need to tell it to use the new IO Manager here? Otherwise it won't actually generate any Parquet files will it?

Reply:
We are instantiating `PudlMixedFormatIOManager` here which uses either of the formats depending on the configuration.